core/capabilities/ccip/configs/evm/chain_writer_test.go

```diff
@@ -0,0 +1,27 @@
+package evm_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/configs/evm"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/assets"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/testutils"
+	evmrelaytypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+func Test_ChainWriterConfig_MarshalUnmarshal(t *testing.T) {
```

Comment:
Shouldn't we cover some other edge cases for this marshall/unmarshall flow?

Reply:
What are some that you're thinking of? I initially added this test because the config we had before was failing to unmarshal due to `TxStrategy` but that has since been removed.

Reply:
nil / empty values / invalid json? if it makes sense of course

Reply:
I think I can add `> 0` gas limit checks and `maxGasPrice > 0` checks, will do that.